module/Kanbanize/src/Kanbanize/Controller/BoardsController.php

```diff
@@ -0,0 +1,204 @@
+use Zend\Filter\StringTrim;
+use Zend\Filter\StripNewlines;
+use Zend\Filter\StripTags;
+use Zend\View\Model\JsonModel;
+use Zend\Validator\InArray;
+use Zend\Validator\NotEmpty;
+use Kanbanize\KanbanizeStream;
+use Kanbanize\Service\KanbanizeService;
+
+
+class BoardsController extends OrganizationAwareController{
+
+	protected static $resourceOptions = ['POST', 'GET'];
+	protected static $collectionOptions= [];
+	/**
+	 * @var StreamService
+	 */
+	private $streamService;
+	/**
+	 * @var KanbanizeAPI
+	 */
+	private $client;
+	/**
+	 * @var KanbanizeService
+	 */
+	private $kanbanizeService;
+
+	public function __construct(OrganizationService $organizationService, StreamService $streamService, KanbanizeAPI $client, KanbanizeService $kanbanizeService){
+		parent::__construct($organizationService);
+		$this->streamService = $streamService;
+		$this->client = $client;
+		$this->kanbanizeService = $kanbanizeService;
+	}
+
+	public function invoke($id, $data){
+		if(is_null($this->identity())) {
+			$this->response->setStatusCode(401);
+			return $this->response;
+		}
+		if(!$this->isAllowed($this->identity(), $this->organization, 'Kanbanize.BoardSettings.create')) {
+			$this->response->setStatusCode(403);
+			return $this->response;
+		}
+		$notEmptyValidator = new NotEmpty();
+		$error = new ErrorJsonModel();
+		if(!(isset($data['projectId']) && $notEmptyValidator->isValid($data['projectId']))){
+			$error->addSecondaryErrors("projectId", ["Missing project id"]);
+		}
+		if(!(isset($data['streamName']) && $notEmptyValidator->isValid($data['streamName']))){
+			$error->addSecondaryErrors("streamName", ["Stream name cannot be empty"]);
+		}
+		if($error->hasErrors()){
+			$error->setCode(400);
+			$error->setDescription('Some parameters are not valid');
+			$this->response->setStatusCode(400);
+			return $error;
+		}
+		$filters = new FilterChain();
+		$filters->attach(new StringTrim())
+			->attach(new StripNewlines())
+			->attach(new StripTags());
+		$streamName = $filters->filter($data['streamName']);
+		$projectId = $data['projectId'];
+		unset($data['streamName']);
+		unset($data['projectId']);
+
+		$statusValidator = new InArray([
+			'haystack' => [
+					Task::STATUS_IDEA,
+					Task::STATUS_OPEN,
+					Task::STATUS_ONGOING,
+					Task::STATUS_COMPLETED,
+					Task::STATUS_ACCEPTED,
+					Task::STATUS_CLOSED
+			]
+		]);
+		$columnMapping = [];
+		foreach($data as $column=>$status){
+			if(!$statusValidator->isValid($status)){
+				$error->addSecondaryErrors($column, ["Invalid status: {$status}"]);
+			}
+		}
+		if($error->hasErrors()){
+			$error->setCode(400);
+			$error->setDescription('Some parameters are not valid');
+			$this->response->setStatusCode(400);
+			return $error;
+		}
+
+		$organization = $this->getOrganizationService()->getOrganization($this->organization->getId());
+		$kanbanizeSettings = $organization->getSetting(Organization::KANBANIZE_KEY_SETTING);
+		$kanbanizeSettings['boards'][$id]['columnMapping'] = $data;
```

Comment:
Non mi è chiaro cosa contenga `$id`. Dovrebbe essere l'id della board in Kanbanize, ma poi lo utilizzi per recuperare lo stream con una `findStream`... non mi torna